bundles/plugins/org.bonitasoft.studio.designer/src/org/bonitasoft/studio/designer/core/bos/WebFormBOSArchiveFileStoreProvider.java

```diff
@@ -74,33 +74,40 @@
     private final CustomPageBarResourceFactory customPageBarResourceFactory;
 
     @Inject
-    public WebFormBOSArchiveFileStoreProvider(final RepositoryAccessor repositoryAccessor, final CustomPageBarResourceFactory customPageBarResourceFactory) {
+    public WebFormBOSArchiveFileStoreProvider(final RepositoryAccessor repositoryAccessor,
+            final CustomPageBarResourceFactory customPageBarResourceFactory) {
         this.repositoryAccessor = repositoryAccessor;
         this.customPageBarResourceFactory = customPageBarResourceFactory;
     }
 
     /*
      * (non-Javadoc)
-     * @see org.bonitasoft.studio.common.repository.provider.IBOSArchiveFileStoreProvider#getFileStoreForConfiguration(org.bonitasoft.studio.model.process.
-     * AbstractProcess, org.bonitasoft.studio.model.configuration.Configuration)
+     * @see org.bonitasoft.studio.common.repository.provider.
+     * IBOSArchiveFileStoreProvider#getFileStoreForConfiguration(org.bonitasoft.
+     * studio.model.process. AbstractProcess,
+     * org.bonitasoft.studio.model.configuration.Configuration)
      */
     @Override
-    public Set<IRepositoryFileStore> getFileStoreForConfiguration(final AbstractProcess process, final Configuration configuration) {
+    public Set<IRepositoryFileStore> getFileStoreForConfiguration(final AbstractProcess process,
+            final Configuration configuration) {
         final Set<IRepositoryFileStore> result = new HashSet<IRepositoryFileStore>();
         final List<FormMapping> allFormMappings = getAllItemsOfType(process, ProcessPackage.Literals.FORM_MAPPING);
         for (final WebPageFileStore fStore : transform(filter(allFormMappings, withInternalType()),
                 formMappingToFileStore())) {
-            result.add(fStore);
-            try {
-                result.addAll(getRelatedFileStore(fStore));
-            } catch (final BarResourceCreationException | IOException e) {
-                BonitaStudioLog.error("Failed to retrieve related Form resoruces", e);
+            if (fStore != null) {
```

Comment:
If we do that it becomes more a features and should not be done in maintenance to avoid regression.